# AASBtrabalho_grupo2020
This repository contains the work acomplished for the bioinformatics course 'Algoritmos de Análise de Sequências Biológicas'. Please notice the ReadMe file that contains important information about the execution of the scripts.

# README
This work contains one main script, Shell, and 14 auxiliary scripts. To start the program run the script Shell. Follow the instructions that appear in the console. Input the number of the option that you wish to do. 
The add_sequence function allows you to add protein sequences, checking their validity.
The ficheiro function reads fasta files exporting to another file the sequence ID and the sequence itself.
The function annotations allows you to request an identifier string, had some features and allows you to update or create those features.
The function search_best_align searches de local database for the most similar sequence.
The freq_symbol function calculates the frequency of the symbols or subsequences in a sequence or in all sequences of the database, returning the subsequences or symbols as keys and the frequencies, as occurrence numbers, represent the values.
The function find_pattern requires a pattern to search for in the database in the sequence selected by the same user, or in all sequences of base data.
The function tree builds a phylogenetic tree using sequences selected by the user, present in the database.
The function multiplealign1 performes multiple alignment for user-selected sequences, by id.
The function prosite allows to evaluate the functional domains of proteins using the ID selected by the user.
The blast function allows the user to perform a blastp from a file fasta with an online database. The output is a file with the blast result.
Hope you enjoy! Done by: Maria Fernanda Vieira PG40963 Maria Inês Faria PG40964 Miguel Martins PG40969

